packages/central-server/src/apiV2/GETHandler/helpers.js

```diff
@@ -107,15 +139,24 @@ export const getQueryOptionsForColumns = (
       const resource = columnName.split('.')[0];
       const recordType = resourceToRecordType(resource);
       if (!recordTypesInQuery.has(recordType)) {
-        const joinCondition = customJoinConditions[recordType] || [
-          `${recordType}.id`,
-          `${baseRecordType}.${getForeignKeyColumnName(recordType)}`,
-        ];
-        multiJoin.push({ joinWith: recordType, joinCondition, joinType });
-        recordTypesInQuery.add(recordType);
+        const join = customJoinConditions[recordType];
+        if (join) {
```

Comment:
Currently you're checking if a `customJoinCondition` exists before running the `addJoin` function. This means it won't generate default joins

I think we can remove all these pre-checks and _just_ run the `addJoin` function, since we run similar checks within that function